gradle/tests.gradle

```diff
@@ -118,7 +118,15 @@ tasks.withType(Test).configureEach {
 }
 
 tasks.test.configure {
-  maxParallelForks = Runtime.getRuntime().availableProcessors()
```

Comment:
- `maxParallelForks` is only used for unit tests. 
- Replay, Weekly and Reference tests have their own gradle task that set the parallelism. You can set the parallelism for go corset workflows by either removing the `*_TESTS_PARALLELISM` env varaible or setting it to 1.
- You can introduce another variable for UNIT_TESTS_PARALLELISM that is disabled for go corset workflows

Reply:
So (surprisingly) the unit tests are the ones causing the problem (they are getting very expensive now).  We cannot have `maxParallelForks` being defined as above when `go-corset` is being used.  That's basically what I'm doing here.  We could have `UNIT_TESTS_PARALLELISM` and, when that is defined, set `maxParallelForks`.  That would work fine I guess, and is pretty comparable to what we've got here.

Reply:
But I do understand you are saying this is redundant:

```
   if ( ...
      System.getenv().containsKey("WEEKLY_TESTS_PARALLELISM") ||
      System.getenv().containsKey("REFERENCE_TESTS_PARALLELISM")) {
```

Because these keys are never used for the `test` task.

Reply:
Ok, I added `UNIT_TESTS_PARALLELISM`